ethpm_cli/commands/install.py

```diff
@@ -111,25 +121,132 @@ def get_package_aliases(package_name: str, config: Config) -> Iterable[Tuple[str
+        all_release_data, installed_package.resolved_version
+    )
+    if on_chain_install_uri != installed_package.resolved_uri:
+        raise InstallError(
+            f"Install URI found on active registry for {installed_package.resolved_package_name}@"
+            f"{installed_package.resolved_version}: {on_chain_install_uri} does not match the "
+            f"install URI found in local lockfile: {installed_package.resolved_uri}."
+        )
+
+    cli_logger.info(
+        f"{len(all_versions)} versions of {installed_package.resolved_package_name} "
+        f"found on the active registry: {all_versions}."
+    )
+    count = 0
+    while True:
+        count += 1
+        target_version = input("Please enter the version you want to install. ")
+        if count > 5:
+            raise InstallError("Max attempts (5) reached. ")
+        elif target_version == installed_package.resolved_version:
+            cli_logger.info(f"Version already installed: {target_version}. ")
+        elif target_version not in all_versions:
+            cli_logger.info(f"Version unavailable: {target_version}. ")
+        else:
+            break
+
+    # Create an updated args/Package for new install
+    updated_args = copy.deepcopy(args)
+    if installed_package.resolved_package_name != args.package:
+        updated_args.alias = args.package
+    updated_args.uri = pluck_release_data(all_release_data, target_version)
+    updated_args.package_version = target_version
+    updated_package = Package(updated_args, config.ipfs_backend)
+
+    # atomic replace
+    with tempfile.TemporaryDirectory() as tmpdir:
+        tmp_ethpm_dir = Path(tmpdir) / ETHPM_PACKAGES_DIR
+        shutil.copytree(config.ethpm_dir, tmp_ethpm_dir)
+        tmp_config = copy.copy(config)
+        tmp_config.ethpm_dir = tmp_ethpm_dir
+        uninstall_package(args.package, tmp_config)
+        install_package(updated_package, tmp_config)
         shutil.rmtree(config.ethpm_dir)
-        tmp_pkg_dir.replace(config.ethpm_dir)
+        tmp_ethpm_dir.replace(config.ethpm_dir)
+
+    cli_logger.info(
+        f"{updated_args.package} successfully updated to version "
+        f"{updated_args.package_version}."
+    )
+
+
+def pluck_release_data(all_release_data: List[str], target_version: str) -> URI:
+    for version, uri in all_release_data:
+        if version == target_version:
+            return uri
+
+
+def uninstall_package(package_name: str, config: Config) -> None:
+    if not is_package_installed(package_name, config):
+        check_for_aliased_package(package_name, config)
         return
 
+    tmp_package_dir = Path(tempfile.mkdtemp()) / ETHPM_PACKAGES_DIR
+    shutil.copytree(config.ethpm_dir, tmp_package_dir)
+    shutil.rmtree(tmp_package_dir / package_name)
+    uninstall_from_ethpm_lock(package_name, (tmp_package_dir / LOCKFILE_NAME))
+
+    shutil.rmtree(config.ethpm_dir)
+    tmp_package_dir.replace(config.ethpm_dir)
```

Comment:
Is it correct that this isn't atomic and needs a context manager? If something happens b/w 245/46 then we just end up with a missing `ethpm_dir`?

Reply:
Yes, but I think you can fix this simply by just removing line 245 since `tmp_package_dir.replace(...)` will overwrite the target directory (thus no need to remove it beforehand).

https://docs.python.org/3/library/pathlib.html#pathlib.Path.replace

Reply:
Yup, I read that and tried, but for some reason when just using `.replace(...)` without the `shutil.rmtree(...)` line - even though the docs say `If target points to an existing file or directory, it will be unconditionally replaced.` - I get an error

Reply:
Ah, I think it's the same way that you can't blindly remove a non-empty directory in Unix.  Maybe worth leaving a comment here to make it clear why they are both necessary.